The job **`x86_64-gnu-llvm-15`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/6852252556/job/18630352245) [(plain)](https://github.com/rust-lang/rust/commit/60d99ab8837e973c0df5fe518a2ed136c8ac0f20/checks/18630352245/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=gavinleroy
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_7d716990-a567-4c05-aac8-b30b2b9a5e2a
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_PATH=/home/runner/work/_temp/_github_workflow/event.json
GITHUB_GRAPHQL_URL=https://api.github.com/graphql
GITHUB_HEAD_REF=proper-depth-check
GITHUB_JOB=pr
GITHUB_PATH=/home/runner/work/_temp/_runner_file_commands/add_path_7d716990-a567-4c05-aac8-b30b2b9a5e2a
GITHUB_REF=refs/pull/117878/merge
GITHUB_REF_NAME=117878/merge
GITHUB_REF_PROTECTED=false
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=beb9ce27dc932606604ccf78ed34c6bbd508572f
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_7d716990-a567-4c05-aac8-b30b2b9a5e2a
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_7d716990-a567-4c05-aac8-b30b2b9a5e2a
GITHUB_TRIGGERING_ACTOR=gavinleroy
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/117878/merge
GITHUB_WORKFLOW_SHA=beb9ce27dc932606604ccf78ed34c6bbd508572f
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_19_X64=/opt/hostedtoolcache/go/1.19.13/x64
---
failures:

---- [ui] tests/ui/impl-trait/auto-trait-coherence.rs#next stdout ----

error in revision `next`: Error: expected failure status (Some(1)) but received status Some(101).
status: exit status: 101
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/impl-trait/auto-trait-coherence.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--cfg" "next" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/impl-trait/auto-trait-coherence.next" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/impl-trait/auto-trait-coherence.next/auxiliary" "-Ztrait-solver=next"
--- stderr -------------------------------
--- stderr -------------------------------
##[error]error: internal compiler error: compiler/rustc_middle/src/ty/sty.rs:1269:21: expected a projection AliasTy; found OpaqueTy
Build completed unsuccessfully in 0:13:03
thread 'rustc' panicked at /checkout/compiler/rustc_errors/src/lib.rs:1661:9:
Box<dyn Any>
stack backtrace:
stack backtrace:
   0:     0x7fe521bdfefc - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h3554aafed5b77072
   1:     0x7fe521c3ab6d - core::fmt::write::hfc3025841d560d9b
   2:     0x7fe521bd4a75 - std::io::Write::write_fmt::h5db36d42f24ce008
   3:     0x7fe521bdfce5 - std::sys_common::backtrace::print::h4454e33eeca39ba6
   4:     0x7fe521be2dca - std::panicking::default_hook::{{closure}}::h3152746e621378b2
   5:     0x7fe521be2ad4 - std::panicking::default_hook::he425c51b29b28760
   6:     0x7fe522833345 - std[740762f69b611507]::panicking::update_hook::<alloc[b7a74c5a3b5a8cde]::boxed::Box<rustc_driver_impl[25ded04b69ad2660]::install_ice_hook::{closure#0}>>::{closure#0}
   7:     0x7fe521be34c5 - std::panicking::rust_panic_with_hook::h477c0e74123886e3
   8:     0x7fe52561c4e6 - std[740762f69b611507]::panicking::begin_panic::<rustc_errors[48b68f01ebf2b576]::ExplicitBug>::{closure#0}
   9:     0x7fe52561bb06 - std[740762f69b611507]::sys_common::backtrace::__rust_end_short_backtrace::<std[740762f69b611507]::panicking::begin_panic<rustc_errors[48b68f01ebf2b576]::ExplicitBug>::{closure#0}, !>
  10:     0x7fe522776c16 - std[740762f69b611507]::panicking::begin_panic::<rustc_errors[48b68f01ebf2b576]::ExplicitBug>
  11:     0x7fe5256bc037 - <rustc_errors[48b68f01ebf2b576]::HandlerInner>::bug::<alloc[b7a74c5a3b5a8cde]::string::String>
  12:     0x7fe5256b974b - <rustc_errors[48b68f01ebf2b576]::Handler>::bug::<alloc[b7a74c5a3b5a8cde]::string::String>
  13:     0x7fe525653df0 - rustc_middle[2dd641c68d405ca0]::util::bug::opt_span_bug_fmt::<rustc_span[d559feae8ff92474]::span_encoding::Span>::{closure#0}
  14:     0x7fe52564d6fc - rustc_middle[2dd641c68d405ca0]::ty::context::tls::with_opt::<rustc_middle[2dd641c68d405ca0]::util::bug::opt_span_bug_fmt<rustc_span[d559feae8ff92474]::span_encoding::Span>::{closure#0}, !>::{closure#0}
  15:     0x7fe52564d6ae - rustc_middle[2dd641c68d405ca0]::ty::context::tls::with_context_opt::<rustc_middle[2dd641c68d405ca0]::ty::context::tls::with_opt<rustc_middle[2dd641c68d405ca0]::util::bug::opt_span_bug_fmt<rustc_span[d559feae8ff92474]::span_encoding::Span>::{closure#0}, !>::{closure#0}, !>
  16:     0x7fe5227771d2 - rustc_middle[2dd641c68d405ca0]::util::bug::bug_fmt
  17:     0x7fe525778b4f - <rustc_middle[2dd641c68d405ca0]::ty::sty::AliasTy>::trait_def_id
  18:     0x7fe525778f50 - <rustc_middle[2dd641c68d405ca0]::ty::sty::AliasTy>::trait_ref
  19:     0x7fe524a6128b - <rustc_trait_selection[aad9f8708ccf84d6]::traits::coherence::AmbiguityCausesVisitor as rustc_trait_selection[aad9f8708ccf84d6]::solve::inspect::analyse::ProofTreeVisitor>::visit_goal
  20:     0x7fe5249703ff - <rustc_infer[312e0aeac7575717]::infer::InferCtxt>::probe::<core[e013c9275820e17]::ops::control_flow::ControlFlow<!>, <rustc_trait_selection[aad9f8708ccf84d6]::solve::inspect::analyse::InspectCandidate>::visit_nested<rustc_trait_selection[aad9f8708ccf84d6]::traits::coherence::AmbiguityCausesVisitor>::{closure#0}>
  21:     0x7fe524a611af - <rustc_trait_selection[aad9f8708ccf84d6]::traits::coherence::AmbiguityCausesVisitor as rustc_trait_selection[aad9f8708ccf84d6]::solve::inspect::analyse::ProofTreeVisitor>::visit_goal
  22:     0x7fe5249703ff - <rustc_infer[312e0aeac7575717]::infer::InferCtxt>::probe::<core[e013c9275820e17]::ops::control_flow::ControlFlow<!>, <rustc_trait_selection[aad9f8708ccf84d6]::solve::inspect::analyse::InspectCandidate>::visit_nested<rustc_trait_selection[aad9f8708ccf84d6]::traits::coherence::AmbiguityCausesVisitor>::{closure#0}>
  23:     0x7fe524a611af - <rustc_trait_selection[aad9f8708ccf84d6]::traits::coherence::AmbiguityCausesVisitor as rustc_trait_selection[aad9f8708ccf84d6]::solve::inspect::analyse::ProofTreeVisitor>::visit_goal
  24:     0x7fe524991b95 - <rustc_infer[312e0aeac7575717]::infer::InferCtxt as rustc_trait_selection[aad9f8708ccf84d6]::solve::inspect::analyse::ProofTreeInferCtxtExt>::visit_proof_tree::<rustc_trait_selection[aad9f8708ccf84d6]::traits::coherence::AmbiguityCausesVisitor>
  25:     0x7fe524a63b3f - rustc_trait_selection[aad9f8708ccf84d6]::traits::coherence::overlap
  26:     0x7fe524a6294b - rustc_trait_selection[aad9f8708ccf84d6]::traits::coherence::overlapping_impls
  27:     0x7fe5248a8433 - <rustc_middle[2dd641c68d405ca0]::traits::specialization_graph::Children as rustc_trait_selection[aad9f8708ccf84d6]::traits::specialize::specialization_graph::ChildrenExt>::insert
  28:     0x7fe52488c71c - <rustc_middle[2dd641c68d405ca0]::traits::specialization_graph::Graph as rustc_trait_selection[aad9f8708ccf84d6]::traits::specialize::specialization_graph::GraphExt>::insert
  29:     0x7fe5248d82f4 - rustc_trait_selection[aad9f8708ccf84d6]::traits::specialize::specialization_graph_provider
  30:     0x7fe52431da79 - rustc_query_impl[ee5ec1d0ef1af45b]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[ee5ec1d0ef1af45b]::query_impl::specialization_graph_of::dynamic_query::{closure#2}::{closure#0}, rustc_middle[2dd641c68d405ca0]::query::erase::Erased<[u8; 8usize]>>
  31:     0x7fe5244e4749 - <rustc_query_impl[ee5ec1d0ef1af45b]::query_impl::specialization_graph_of::dynamic_query::{closure#2} as core[e013c9275820e17]::ops::function::FnOnce<(rustc_middle[2dd641c68d405ca0]::ty::context::TyCtxt, rustc_span[d559feae8ff92474]::def_id::DefId)>>::call_once
  32:     0x7fe5246c356e - rustc_query_system[ae73670ae956f036]::query::plumbing::try_execute_query::<rustc_query_impl[ee5ec1d0ef1af45b]::DynamicConfig<rustc_query_system[ae73670ae956f036]::query::caches::DefaultCache<rustc_span[d559feae8ff92474]::def_id::DefId, rustc_middle[2dd641c68d405ca0]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[ee5ec1d0ef1af45b]::plumbing::QueryCtxt, false>
  33:     0x7fe52460ed87 - rustc_query_impl[ee5ec1d0ef1af45b]::query_impl::specialization_graph_of::get_query_non_incr::__rust_end_short_backtrace
  34:     0x7fe5232bcbcc - rustc_hir_analysis[54bffc647343d]::coherence::coherent_trait
  35:     0x7fe524306baa - rustc_query_impl[ee5ec1d0ef1af45b]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[ee5ec1d0ef1af45b]::query_impl::coherent_trait::dynamic_query::{closure#2}::{closure#0}, rustc_middle[2dd641c68d405ca0]::query::erase::Erased<[u8; 0usize]>>
  36:     0x7fe5244a9529 - <rustc_query_impl[ee5ec1d0ef1af45b]::query_impl::coherent_trait::dynamic_query::{closure#2} as core[e013c9275820e17]::ops::function::FnOnce<(rustc_middle[2dd641c68d405ca0]::ty::context::TyCtxt, rustc_span[d559feae8ff92474]::def_id::DefId)>>::call_once
  37:     0x7fe5246ac1a4 - rustc_query_system[ae73670ae956f036]::query::plumbing::try_execute_query::<rustc_query_impl[ee5ec1d0ef1af45b]::DynamicConfig<rustc_query_system[ae73670ae956f036]::query::caches::DefaultCache<rustc_span[d559feae8ff92474]::def_id::DefId, rustc_middle[2dd641c68d405ca0]::query::erase::Erased<[u8; 0usize]>>, false, false, false>, rustc_query_impl[ee5ec1d0ef1af45b]::plumbing::QueryCtxt, false>
  38:     0x7fe52460e597 - rustc_query_impl[ee5ec1d0ef1af45b]::query_impl::coherent_trait::get_query_non_incr::__rust_end_short_backtrace
  39:     0x7fe52325b53d - rustc_hir_analysis[54bffc647343d]::check_crate
  41:     0x7fe524326053 - rustc_query_impl[ee5ec1d0ef1af45b]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[ee5ec1d0ef1af45b]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[2dd641c68d405ca0]::query::erase::Erased<[u8; 1usize]>>
  42:     0x7fe524608791 - <rustc_query_impl[ee5ec1d0ef1af45b]::query_impl::analysis::dynamic_query::{closure#2} as core[e013c9275820e17]::ops::function::FnOnce<(rustc_middle[2dd641c68d405ca0]::ty::context::TyCtxt, ())>>::call_once
  43:     0x7fe524656a3a - rustc_query_system[ae73670ae956f036]::query::plumbing::try_execute_query::<rustc_query_impl[ee5ec1d0ef1af45b]::DynamicConfig<rustc_query_system[ae73670ae956f036]::query::caches::SingleCache<rustc_middle[2dd641c68d405ca0]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[ee5ec1d0ef1af45b]::plumbing::QueryCtxt, false>
  44:     0x7fe52460cf9d - rustc_query_impl[ee5ec1d0ef1af45b]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  44:     0x7fe52460cf9d - rustc_query_impl[ee5ec1d0ef1af45b]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  45:     0x7fe522813e51 - <rustc_middle[2dd641c68d405ca0]::ty::context::GlobalCtxt>::enter::<rustc_driver_impl[25ded04b69ad2660]::run_compiler::{closure#1}::{closure#2}::{closure#6}, core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>>
  46:     0x7fe5228c335a - <rustc_interface[3c421cfc13622318]::interface::Compiler>::enter::<rustc_driver_impl[25ded04b69ad2660]::run_compiler::{closure#1}::{closure#2}, core[e013c9275820e17]::result::Result<core[e013c9275820e17]::option::Option<rustc_interface[3c421cfc13622318]::queries::Linker>, rustc_span[d559feae8ff92474]::ErrorGuaranteed>>
  47:     0x7fe5228511f5 - rustc_span[d559feae8ff92474]::set_source_map::<core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>, rustc_interface[3c421cfc13622318]::interface::run_compiler<core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>, rustc_driver_impl[25ded04b69ad2660]::run_compiler::{closure#1}>::{closure#0}::{closure#0}>
  48:     0x7fe522837f8f - <scoped_tls[2c27d1b2750614a4]::ScopedKey<rustc_span[d559feae8ff92474]::SessionGlobals>>::set::<rustc_interface[3c421cfc13622318]::util::run_in_thread_pool_with_globals<rustc_interface[3c421cfc13622318]::interface::run_compiler<core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>, rustc_driver_impl[25ded04b69ad2660]::run_compiler::{closure#1}>::{closure#0}, core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>>::{closure#0}, core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>>
  49:     0x7fe522851d2c - rustc_span[d559feae8ff92474]::create_session_globals_then::<core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>, rustc_interface[3c421cfc13622318]::util::run_in_thread_pool_with_globals<rustc_interface[3c421cfc13622318]::interface::run_compiler<core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>, rustc_driver_impl[25ded04b69ad2660]::run_compiler::{closure#1}>::{closure#0}, core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>>::{closure#0}>
  50:     0x7fe52281844e - std[740762f69b611507]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[3c421cfc13622318]::util::run_in_thread_with_globals<rustc_interface[3c421cfc13622318]::util::run_in_thread_pool_with_globals<rustc_interface[3c421cfc13622318]::interface::run_compiler<core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>, rustc_driver_impl[25ded04b69ad2660]::run_compiler::{closure#1}>::{closure#0}, core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>>::{closure#0}, core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>>
  51:     0x7fe52282eefe - std[740762f69b611507]::panicking::try::<core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>, core[e013c9275820e17]::panic::unwind_safe::AssertUnwindSafe<<std[740762f69b611507]::thread::Builder>::spawn_unchecked_<rustc_interface[3c421cfc13622318]::util::run_in_thread_with_globals<rustc_interface[3c421cfc13622318]::util::run_in_thread_pool_with_globals<rustc_interface[3c421cfc13622318]::interface::run_compiler<core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>, rustc_driver_impl[25ded04b69ad2660]::run_compiler::{closure#1}>::{closure#0}, core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>>::{closure#0}, core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>>::{closure#1}::{closure#0}>>
  52:     0x7fe522819820 - <<std[740762f69b611507]::thread::Builder>::spawn_unchecked_<rustc_interface[3c421cfc13622318]::util::run_in_thread_with_globals<rustc_interface[3c421cfc13622318]::util::run_in_thread_pool_with_globals<rustc_interface[3c421cfc13622318]::interface::run_compiler<core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>, rustc_driver_impl[25ded04b69ad2660]::run_compiler::{closure#1}>::{closure#0}, core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>>::{closure#0}, core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[e013c9275820e17]::result::Result<(), rustc_span[d559feae8ff92474]::ErrorGuaranteed>>::{closure#1} as core[e013c9275820e17]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  54:     0x7fe5219bcac3 - <unknown>
  55:     0x7fe521a4ea40 - <unknown>
  56:                0x0 - <unknown>


note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.76.0-nightly (beb9ce27d 2023-11-13) running on x86_64-unknown-linux-gnu

note: compiler flags: -Z threads=1 -Z simulate-remapped-rust-src-base=/rustc/FAKE_PREFIX -Z translate-remapped-path-to-local-path=no -Z ignore-directory-in-diagnostics-source-blocks=/cargo -C codegen-units=1 -Z ui-testing -Z deduplicate-diagnostics=no -Z write-long-types-to-disk=no -C strip=debuginfo -C prefer-dynamic -C rpath -C debuginfo=0 -Z trait-solver=next
query stack during panic:
query stack during panic:
#0 [specialization_graph_of] building specialization graph of trait `AnotherTrait`
#1 [coherent_trait] coherence checking all impls of trait `AnotherTrait`
end of query stack
error: aborting due to previous error
------------------------------------------

```

</details>

--- a/compiler/rustc_trait_selection/src/traits/coherence.rs
+++ b/compiler/rustc_trait_selection/src/traits/coherence.rs
@@ -20,6 +20,7 @@ use crate::traits::{
 };
 use rustc_data_structures::fx::FxIndexSet;
 use rustc_errors::Diagnostic;
+use rustc_hir::def::DefKind;
 use rustc_hir::def_id::{DefId, LOCAL_CRATE};
 use rustc_infer::infer::{DefineOpaqueTypes, InferCtxt, TyCtxtInferExt};
 use rustc_infer::traits::{util, TraitEngine};
@@ -1002,7 +1003,12 @@ impl<'a, 'tcx> ProofTreeVisitor<'tcx> for AmbiguityCausesVisitor<'a> {
         // and then prove the resulting predicate as a nested goal.
         let trait_ref = match predicate.kind().no_bound_vars() {
             Some(ty::PredicateKind::Clause(ty::ClauseKind::Trait(tr))) => tr.trait_ref,
-            Some(ty::PredicateKind::Clause(ty::ClauseKind::Projection(proj))) => {
+            Some(ty::PredicateKind::Clause(ty::ClauseKind::Projection(proj)))
+                if matches!(
+                    infcx.tcx.def_kind(proj.projection_ty.def_id),
+                    DefKind::AssocTy | DefKind::AssocConst
+                ) =>
+            {
                 proj.projection_ty.trait_ref(infcx.tcx)
             }
             _ => return ControlFlow::Continue(()),

--- a/tests/ui/impl-trait/auto-trait-coherence.next.stderr
+++ b/tests/ui/impl-trait/auto-trait-coherence.next.stderr
@@ -6,6 +6,8 @@ LL | impl<T: Send> AnotherTrait for T {}
 ...
 LL | impl AnotherTrait for D<OpaqueType> {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `D<OpaqueType>`
+   |
+   = note: upstream crates may add a new impl of trait `std::marker::Send` for type `OpaqueType` in future versions
 
 error: aborting due to previous error
 

